File: backend/java/Vehicle_Configurator/src/main/java/com/example/demo/entities/InvoiceDetail.java
package com.example.demo.entities;

import jakarta.persistence.*;

@Entity
@Table(name = "invoice_detail")
public class InvoiceDetail {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "invoice_detail_id")
    private int invoiceDetailId;

    @ManyToOne
    @JoinColumn(name = "invoice_id")
    private Invoice invoice;

    // Original Component
    @ManyToOne
    @JoinColumn(name = "component_id")
    private Component component;

    // Selected Alternative Component
    @ManyToOne
    @JoinColumn(name = "alternate_component_id")
    private Component alternateComponent;

    @Column(name = "delta_price")
    private double deltaPrice;

    public InvoiceDetail() {
    }

    public int getInvoiceDetailId() {
        return invoiceDetailId;
    }

    public void setInvoiceDetailId(int invoiceDetailId) {
        this.invoiceDetailId = invoiceDetailId;
    }

    public Invoice getInvoice() {
        return invoice;
    }

    public void setInvoice(Invoice invoice) {
        this.invoice = invoice;
    }

    public Component getComponent() {
        return component;
    }

    public void setComponent(Component component) {
        this.component = component;
    }

    public Component getAlternateComponent() {
        return alternateComponent;
    }

    public void setAlternateComponent(Component alternateComponent) {
        this.alternateComponent = alternateComponent;
    }

    public double getDeltaPrice() {
        return deltaPrice;
    }

    public void setDeltaPrice(double deltaPrice) {
        this.deltaPrice = deltaPrice;
    }
}

File: backend-dotnet/Models/InvoiceDetail.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend_dotnet.Models
{
    [Table("invoice_detail")]
    public class InvoiceDetail
    {
        [Key]
        [Column("invoice_detail_id")]
        public int InvoiceDetailId { get; set; }

        [Column("invoice_id")]
        public int InvoiceId { get; set; }

        [ForeignKey(nameof(InvoiceId))]
        public Invoice? Invoice { get; set; }

        // Original Component
        [Column("component_id")]
        public int ComponentId { get; set; }

        [ForeignKey(nameof(ComponentId))]
        public LoginRequest? Component { get; set; }

        // Selected Alternate Component
        [Column("alternate_component_id")]
        public int AlternateComponentId { get; set; }

        [ForeignKey(nameof(AlternateComponentId))]
        public LoginRequest? AlternateComponent { get; set; }

        [Column("delta_price")]
        public double DeltaPrice { get; set; }
    }
}
